### PR TITLE
Upgrade to lucene-8.5.0-snapshot-3333ce7da6d

### PR DESCRIPTION
**Pull request by mayya-sharipova**

### CODE REVIEW

**Comment by jpountz** [comment]
I'd like @romseygeek to have a look at the highlighting/intervals changes and @nknize to look ah the change to the classpath for the server module.

File: server/src/main/java/org/elasticsearch/index/query/IntervalsSourceProvider.java [hunk 1]
```diff
@@ -797,8 +797,8 @@ public IntervalsSource getSource(QueryShardContext context, MappedFieldType fiel
             BytesRef normalizedTerm = analyzer.normalize(fieldType.name(), term);
             FuzzyQuery fq = new FuzzyQuery(new Term(fieldType.name(), normalizedTerm),
                 fuzziness.asDistance(term), prefixLength, 128, transpositions);
-            CompiledAutomaton ca = new CompiledAutomaton(fq.toAutomaton());
-            source = XIntervals.multiterm(ca, term);
+            CompiledAutomaton[] automata = fq.getAutomata();
+            source = XIntervals.multiterm(automata[automata.length - 1], term);
             if (useField != null) {
```
Comment (romseygeek) [inline server/src/main/java/org/elasticsearch/index/query/IntervalsSourceProvider.java]: I think we can remove `XIntervals` now that LUCENE-9050 is released - just use plain `Intervals` instead.

File: server/src/main/java/org/apache/lucene/search/uhighlight/CustomFieldHighlighter.java [hunk 2]
```diff
@@ -76,4 +82,84 @@
...
+            passageQueue.offer(passage);
+            if (passageQueue.size() > maxPassages) {
+                passage = passageQueue.poll();
+                passage.reset();
+            } else {
+                passage = new Passage();
+            }
+        }
+        return passage;
+    }
```
Comment (romseygeek) [inline server/src/main/java/org/apache/lucene/search/uhighlight/CustomFieldHighlighter.java]: This just overrides LUCENE-9093 and preserves the previous behaviour, correct? +1 if so

**Review by romseygeek — CHANGES_REQUESTED** [review]
There's one more change I think we can make, removing `XIntervals`.  LGTM otherwise.

File: server/src/main/java/org/apache/lucene/search/uhighlight/CustomFieldHighlighter.java (hunk 2 above)
Comment (mayya-sharipova) [reply server/src/main/java/org/apache/lucene/search/uhighlight/CustomFieldHighlighter.java]: exactly, it just preserves the previous behaviour.

File: server/src/main/java/org/elasticsearch/index/query/IntervalsSourceProvider.java (hunk 1 above)
Comment (mayya-sharipova) [reply server/src/main/java/org/elasticsearch/index/query/IntervalsSourceProvider.java]: @romseygeek Thanks for the comment. Looks like `Intervals.multiterm` expects `Automaton`, while a `FuzzyQuery` can only produce `CompiledAutomaton`.  Or is there any other way to use `Intervals.multiterm`?

Comment (romseygeek) [reply server/src/main/java/org/elasticsearch/index/query/IntervalsSourceProvider.java]: Hm, maybe we need to put a `fuzzy` method directly on `Intervals` then.  Let's leave `XIntervals` in there for the moment, I'll open a lucene ticket to clean this up.

**Comment by nknize** [comment]
The classpath for removing the spatial module LGTM. Glad we had already removed the dependency on the `MortonEncoder`. Seems to have made the spatial removal a lot easier!  Thx @mayya-sharipova

**Comment by mayya-sharipova** [comment]
@jpountz Do you have any other feedback, or this can be merged?

**Review by jpountz — APPROVED** [review]
Thanks Mayya!

**Pull request merged**